File: utility_tools/test_c.r
#' Runs through the functions and outputs results so they can be compared to the C code outputs
source("NG_FWI.r")
library(ggplot2)
library(data.table)

library(lubridate)
df_wx <- as.data.table(read.csv("./data/wx_hourly.csv"))
result <- hFWI(df_wx, timezone = -6)
save_csv(result, "./result.csv")

# # create data file from cffdrs R package file
# test_hffmc <- read.csv("test_hffmc.csv")
# df <- data.table(test_hffmc)
# df[, lat := df_wx$lat[[1]]]
# df[, long := df_wx$long[[1]]]
# write.table(df[, c("lat", "long", "yr", "mon", "day", "hr", "temp", "rh", "ws", "rh")], "./data/test_hffmc.csv", quote = FALSE, row.names = FALSE, sep = ",")


df <- as.data.table(read.csv("./data/test_hffmc.csv"))

result_hffmc <- hFWI(df, timezone = -6)
save_csv(result_hffmc, "./result_hffmc.csv")

source("make_daily.r")
df_wx <- as.data.table(read.csv("./data/wx_hourly.csv"))
df <- hourly_to_daily(df_wx)
save_csv(df, "./data/wx_daily.csv")


source("make_minmax.r")
df_wx <- as.data.table(read.csv("./data/wx_daily.csv"))
df <- daily_to_minmax(df_wx)
save_csv(df, "./data/wx_minmax.csv")



source("make_hourly.r")
wx_minmax <- as.data.table(read.csv("./data/wx_minmax.csv"))
df <- minmax_to_hourly(wx_minmax, timezone = -6)
df_hourly <- copy(df)
save_csv(df, "./data/wx_diurnal.csv")


wx_diurnal <- as.data.table(read.csv("./data/wx_diurnal.csv"))
result3 <- hFWI(wx_diurnal, timezone = -6)
save_csv(result3, "./result3.csv")


wx_windy <- as.data.table(read.csv("./data/wx_windy.csv"))
result4 <- hFWI(wx_windy, timezone = -6)
save_csv(result4, "./result4.csv")

save_csv(hFWI(as.data.table(read.csv("./data/wx_rh100.csv")), timezone = -6), "./result5.csv")
save_csv(hFWI(as.data.table(read.csv("./data/wx_rh0.csv")), timezone = -6), "./result6.csv")

File: utility_tools/test_c.py
import pandas as pd

import make_daily
import make_hourly
import make_minmax
import NG_FWI
from util import save_csv


def run_tests():
    df_wx = pd.read_csv("./data/wx_hourly.csv")
    result = NG_FWI.hFWI(df_wx, timezone=-6)
    save_csv(result, "./result_py.csv")

    df_wx = pd.read_csv("./data/wx_hourly.csv")
    test_hffmc = pd.read_csv("test_hffmc.csv")
    df = test_hffmc.copy()
    df["lat"] = df_wx["lat"].iloc[0]
    df["long"] = df_wx["long"].iloc[0]
    COLUMN_SYNONYMS = {"yr": "year", "hr": "hour", "ws": "wind", "prec": "rain"}
    df = df.rename(columns=COLUMN_SYNONYMS)
    df = df[["lat", "long", "year", "mon", "day", "hour", "temp", "rh", "wind", "rain"]]
    save_csv(df, "./data/test_hffmc_py.csv")
    df = pd.read_csv("./data/test_hffmc_py.csv")

    result_hffmc = NG_FWI.hFWI(df, timezone=-6)
    save_csv(result_hffmc, "./result_hffmc_py.csv")

    df_wx = pd.read_csv("./data/wx_hourly.csv")
    df = make_daily.hourly_to_daily(df_wx)
    df = df[["lat", "long", "year", "mon", "day", "hour", "temp", "rh", "wind", "rain"]]
    save_csv(df, "./data/wx_daily_py.csv")

    df_wx = pd.read_csv("./data/wx_daily_py.csv")
    df = make_minmax.daily_to_minmax(df_wx)
    df = df[
        [
            "lat",
            "long",
            "year",
            "mon",
            "day",
            "hour",
            "temp_min",
            "temp_max",
            "rh_min",
            "rh_max",
            "ws_min",
            "ws_max",
            "rain",
        ]
    ]
    save_csv(df, "./data/wx_minmax_py.csv")

    wx_minmax = pd.read_csv("./data/wx_minmax_py.csv")
    df = make_hourly.minmax_to_hourly(wx_minmax, timezone=-6)
    df = df[["lat", "long", "year", "mon", "day", "hour", "temp", "rh", "wind", "rain"]]
    save_csv(df, "./data/wx_diurnal_py.csv")

    wx_diurnal = pd.read_csv("./data/wx_diurnal_py.csv")
    result3 = NG_FWI.hFWI(wx_diurnal, timezone=-6)
    save_csv(result3, "./result3_py.csv")

    wx_windy = pd.read_csv("./data/wx_windy.csv")
    result4 = NG_FWI.hFWI(wx_windy, timezone=-6)
    save_csv(result4, "result4_py.csv")

    save_csv(
        NG_FWI.hFWI(pd.read_csv("./data/wx_rh100.csv"), timezone=-6), "./result5_py.csv"
    )
    save_csv(
        NG_FWI.hFWI(pd.read_csv("./data/wx_rh0.csv"), timezone=-6), "./result6_py.csv"
    )


if __name__ == "__main__":
    run_tests()
